force-app/main/default/classes/DMLOperations.cls

```diff
@@ -181,14 +253,20 @@ public with sharing class DMLOperations {
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
 		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
+        for (Contact contact : contactsToUpsert){
+        // Extract the Account Name from Contact's LastName
+        String accountName = contact.LastName;
 
 		// Get the account based on the name or create one if it doesn't exist
+            Account account = upsertAccount(accountName);
```

Comment:
This is DML within a loop again because it is calling the other method, which does DML and SOQL.